### DISCUSSION TITLE
Product Attribute Import

### DISCUSSION THREAD

**Discussion by paultvis** [atrocore/atrocore#487] [Q&A]
Hi,

We are on the last step of migrating from Akeneo to Atro, but are having issues importing the attribute values for products. The behavior seems erratic, and the imports generate lots of  'Attribute '<attribute_name>' for the channel Global is already linked errors. We have about 40k products to upload, with 650k attribute values. We have broken the imports down to 1k products, then the associated attribute values, which is about 17k. I have tried breaking the 17k into 1k imports but the errors still persist. 

With the same data set, on some import attemps we get 10s of attribute values imported, other runs get a few hundred. There are no errors being raised. In the database we can see all the attribute values are there, they are just not populated with values. 

Any suggestions why this would be happening?

Thanks

Paul

**Comment by atrolex** [comment]
Please contact our support to get paid service. It is hard to help you quickly without knowing details.

**Reply by paultvis** [reply]
Hi Alex,

We did resolve this, along with a few other minor issues. The issue here was that the enpty attribute values are created when the product is created if a family is added to the product. We were not using the IDs from the empty attribute values in the import and atro was not tying the existing attribute values to the product import using just the SKU and was trying to create duplicates. Once we got the ids for the attribute values to be imported the issue was resolved.

The upshot from this is we now have a working akeneo to atropim migration path which takes over all entities (products, families, attributes, groups, catalogs, categories, etc). For our catalogue with 45k products, it takes about 3 days to run end to end. One day extracting and preparing the data and about 2 days running imports (we have kept this at batches of 1k products as said as I was not sure if it was viable to just run a full 45k product import then 650k product attributes in single runs). 

Regards,

Paul

**Reply by atrolex** [reply]
Ok, my congratulations  :) This was definitely not an easy task.
